src/main/java/seedu/address/logic/parser/ParserUtil.java

```diff
@@ -231,12 +235,35 @@ public static Date parseDate(String date) throws ParseException {
      * Parses a {@code String description} into a {@code Description}.
      * Leading and trailing whitespaces will be trimmed.
      */
-    public static IterationDescription parseIterationDescription(String description) throws ParseException {
+    public static IterationDescription parseIterationDescription(String description) {
         requireNonNull(description);
         String trimmedDescription = description.trim();
         return new IterationDescription(trimmedDescription);
     }
 
+    /**
+     * Parses a {@code String imagePath} into a {@code ImagePath}.
+     * Leading and trailing whitespaces will be trimmed.
+     */
+    public static ImagePath parseImagePath(String imagePath) throws ParseException {
+        requireNonNull(imagePath);
+        String trimmedImagePath = imagePath.trim();
+        Path srcPath = Path.of(trimmedImagePath);
+        if (Files.exists(srcPath)) {
+            try {
+                System.out.println(Files.probeContentType(srcPath));
+                if (Files.probeContentType(srcPath).split("/")[0].equals("image")) {
+                    return new ImagePath(trimmedImagePath);
+                }
+                throw new ParseException(Messages.MESSAGE_NOT_AN_IMAGE);
+            } catch (IOException e) {
+                e.printStackTrace();
+            }
+        }
+        throw new ParseException(Messages.MESSAGE_NONEXISTENT_IMAGE_PATH);
```

Comment:
Should we use a guard clause to make this cleaner? [Code Quality - Making the happy path prominent](https://nus-cs2103-ay2223s1.github.io/website/se-book-adapted/chapters/codeQuality.html#use-name-to-explain:~:text=Advanced-,Make%20the%20Happy%20Path%20Prominent,-Can%20improve%20code)